File: java/aholo-sdk-lux3d/src/test/java/com/manycoreapis/sdk/lux3d/resources/NewApisTest.java
package com.manycoreapis.sdk.lux3d.resources;

import com.manycoreapis.sdk.core.AholoClientConfig;
import com.manycoreapis.sdk.core.AholoGatewayClient;
import com.manycoreapis.sdk.lux3d.model.PartSplitCreateParams;
import com.manycoreapis.sdk.lux3d.model.TaskListParams;
import com.manycoreapis.sdk.lux3d.model.TaskPagedList;
import org.junit.jupiter.api.Test;

import java.util.Arrays;
import java.util.LinkedHashMap;
import java.util.Map;

import static org.junit.jupiter.api.Assertions.assertEquals;

final class NewApisTest {
    @Test
    void partSplitUsesExpectedEndpoint() {
        FakeGateway gateway = new FakeGateway();
        long taskId = new PartSplitResource(gateway, "/lux3d/v1")
                .create(PartSplitCreateParams.builder().glbUrl("https://example.com/model.glb").build());
        assertEquals(42L, taskId);
        assertEquals("POST", gateway.method);
        assertEquals("/lux3d/v1/part-split/task/create", gateway.path);
    }

    @Test
    void taskListMapsQueryAndResponse() {
        FakeGateway gateway = new FakeGateway();
        TaskPagedList result = new TasksResource(gateway, "/global/lux3d/v1").list(
                TaskListParams.builder().page(2).pageSize(10).status(3).startTime(100).endTime(200).build());
        assertEquals("/global/lux3d/v1/generate/task/list", gateway.path);
        assertEquals(2, gateway.query.get("page"));
        assertEquals(10, gateway.query.get("pagesize"));
        assertEquals(3, gateway.query.get("status"));
        assertEquals(100L, gateway.query.get("starttime"));
        assertEquals(200L, gateway.query.get("endtime"));
        assertEquals(1, result.total());
        assertEquals(7L, result.items().get(0).taskId());
    }

    private static final class FakeGateway extends AholoGatewayClient {
        private String method;
        private String path;
        private Map<String, Object> query;

        private FakeGateway() {
            super(AholoClientConfig.of("test-key", "cn"));
        }

        @Override
        public Map<String, Object> gatewayRequest(
                String method, String path, Map<String, Object> query, Map<String, String> headers, Object body) {
            this.method = method;
            this.path = path;
            this.query = query;
            Map<String, Object> response = new LinkedHashMap<String, Object>();
            response.put("c", "0");
            response.put("m", "");
            if (path.endsWith("/part-split/task/create")) {
                response.put("d", 42);
            } else {
                Map<String, Object> item = new LinkedHashMap<String, Object>();
                item.put("taskId", 7L);
                item.put("status", 3);
                item.put("created", 100L);
                item.put("lastModified", 200L);
                Map<String, Object> data = new LinkedHashMap<String, Object>();
                data.put("items", Arrays.<Map<String, Object>>asList(item));
                data.put("total", 1);
                data.put("page", 2);
                data.put("pageSize", 10);
                response.put("d", data);
            }
            return response;
        }
    }
}

File: typescript/packages/aholo-sdk-lux3d/test/new-apis.test.ts
import assert from 'node:assert/strict';
import test from 'node:test';

import type { AholoGatewayClient } from '@manycore/aholo-sdk-core';

import { PartSplitResource } from '../src/resources/part-split.js';
import { TasksResource } from '../src/resources/tasks.js';

test('part split uses the expected endpoint and returns the task id', async () => {
  let request: unknown;
  const gateway = {
    gatewayRequest: async (options: unknown) => {
      request = options;
      return { c: '0', m: '', d: 42 };
    },
  } as AholoGatewayClient;

  const resource = new PartSplitResource(gateway, 'cn');
  assert.equal(await resource.create({ glbUrl: 'https://example.com/model.glb' }), 42);
  assert.deepEqual(request, {
    method: 'POST',
    path: '/lux3d/v1/part-split/task/create',
    body: { glbUrl: 'https://example.com/model.glb' },
    signal: undefined,
  });
});

test('task list maps camelCase filters and omits unset values', async () => {
  const requests: unknown[] = [];
  const gateway = {
    gatewayRequest: async (options: unknown) => {
      requests.push(options);
      return { c: '0', m: '', d: { items: [], total: 0, page: 2, pageSize: 10 } };
    },
  } as AholoGatewayClient;

  const resource = new TasksResource(gateway, 'com');
  const result = await resource.list({ page: 2, pageSize: 10, status: 3, startTime: 100, endTime: 200 });
  assert.equal(result.total, 0);
  assert.deepEqual(requests[0], {
    method: 'GET',
    path: '/global/lux3d/v1/generate/task/list',
    query: { page: '2', pagesize: '10', status: '3', starttime: '100', endtime: '200' },
    signal: undefined,
  });

  await resource.list();
  assert.deepEqual(requests[1], {
    method: 'GET',
    path: '/global/lux3d/v1/generate/task/list',
    query: {},
    signal: undefined,
  });
});
